alerting/src/main/kotlin/org/opensearch/alerting/DocumentReturningMonitorRunner.kt

```diff
@@ -238,7 +322,10 @@ object DocumentReturningMonitorRunner : MonitorRunner {
         query: DocLevelQuery,
         index: String
     ): Set<String> {
-        val count: Int = docExecutionCtx.lastRunContext["shards_count"] as Int
+        // TODO DRAFT: Is it better to get shards_count from updatedLastRunContext as opposed to lastRunContext?
+        //  The number of shards could changing between monitor executions
+//        val count: Int = docExecutionCtx.lastRunContext["shards_count"] as Int
+        val count: Int = docExecutionCtx.updatedLastRunContext["shards_count"] as Int
```

Comment:
This should be right and since this is being done, make sure when passing the `prevSeqNo` in `searchShard`, modify `docExecutionCtx.lastRunContext[shard].toString().toLongOrNull()` to `docExecutionCtx.lastRunContext[shard]?.toString().toLongOrNull()` as `shard` might not exist in it.